cmd/strawToBedpe/strawToBedpe.go

```diff
@@ -0,0 +1,70 @@
+// Command Group: "Data Conversion"
+
+// Convert the output of the aidenlab's straw command to a bedpe format
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/bed/bedpe"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/hic"
+	"log"
+)
+
+func strawToBedpe(strawFile string, outFile string, chrom string, binSize int, interChrom string) {
+	var thisRec bedpe.BedPe
+	var err error
+
+	straw := hic.GoReadToChan(strawFile)
+	out := fileio.EasyCreate(outFile)
+
+	if interChrom == "" {
+		for s := range straw {
+			thisRec = bedpe.BedPe{A: bed.Bed{Chrom: chrom, ChromStart: s.Bin1Start, ChromEnd: s.Bin1Start + binSize, Score: s.ContactScore, FieldsInitialized: 8}, B: bed.Bed{Chrom: chrom, ChromStart: s.Bin2Start, ChromEnd: s.Bin2Start + binSize, Score: s.ContactScore, FieldsInitialized: 8}}
+			bedpe.WriteToFileHandle(out, thisRec)
+		}
+	} else {
+		for s := range straw {
+			thisRec = bedpe.BedPe{A: bed.Bed{Chrom: chrom, ChromStart: s.Bin1Start, ChromEnd: s.Bin1Start + binSize, Score: s.ContactScore, FieldsInitialized: 8}, B: bed.Bed{Chrom: interChrom, ChromStart: s.Bin2Start, ChromEnd: s.Bin2Start + binSize, Score: s.ContactScore, FieldsInitialized: 8}}
+			bedpe.WriteToFileHandle(out, thisRec)
+		}
+	}
+
+	err = out.Close()
+	exception.PanicOnErr(err)
+}
+
+func usage() {
+	fmt.Print(
+		"strawToBedpe - convert the output of the aidenlab's straw command to a bedpe format." +
+			"\n" +
+			"Usage:\n" +
+			"\n" +
+			"options:\n")
+	flag.PrintDefaults()
+}
+
+func main() {
+	var expectedNumArgs int = 4
+	var binSize *int = flag.Int("binSize", 5000, "A binSize must be provided for the resolution of the straw file output if it is not 5000.")
```

Comment:
what is the behavior if the input bin size is different than the straw bin size? Is there any way to autodetect bin size from the straw file and keep it in the straw struct?

Reply:
maybe a better way of phrasing this question is: does the input binSize from the option have to equal the bin size from the straw file?

Reply:
yes it does if you want the data to be accurate to the actual contact information

Reply:
Should the program throw an error if the actual vs input bin sizes aren't equal?

Reply:
The straw file doesn't contain any information about window size so there's no straight forward way to check for it.